packages/navigation/src/UILargeTitleHeader/useScrollHandler/useOnEndDrag.tsx

```diff
@@ -0,0 +1,275 @@
+                        if (shift.value > 0) {
+                            shift.value = withSpring(0, {
+                                overshootClamping: true,
+                            });
+                            return;
+                        }
+                        /**
+                         * Next we look if the header became visible.
+                         * Here we handle when it doesn't.
+                         * That means that we should wait until `onMomentumEnd` fired
+                         * as ScrollView will continue to fire regular `onScroll` events until it.
+                         */
+                        if (event.contentOffset.y > 0) {
+                            if (ctx != null) {
+                                ctx.continueResetOnMomentumEnd = true;
+                            }
+
+                            return;
+                        }
+
+                        movementHandlers.onUpwardDeceleration(
+                            runOnUIPlatformSelect({
+                                ios: -1 * event.velocity.y,
+                                android: event.velocity.y * 5,
+                                default: event.velocity.y,
+                            }),
+                            runOnUIPlatformSelect({
+                                ios: 500,
+                                default: 100,
+                            }),
+                            ctx,
+                        );
+
+                        return;
+                    }
+                    /**
+                     * Just fun story about scrolling on Android:
+                     * When you scrolling with your finger imagine the moment
+                     * when you release it, you actually can be surprised, but
+                     * actually there could be a backward movement that your pad
+                     * of a finger can produce, when it released, so at the end
+                     * you can get slight velocity in backward direction.
+                     * In oreder to reduce it treat such values as the ones
+                     * without inertia.
+                     * We achive this by accepting velocity values only more than 0.6.
+                     */
+                    const isToBottomMotion = runOnUIPlatformSelect({
+                        android: event.velocity.y < -0.6,
+                        default: event.velocity.y > 0,
+                    });
+                    if (isToBottomMotion) {
+                        // Nothing to do, regular scroll
+                        if (event.contentOffset.y > 0) {
+                            return;
+                        }
+
+                        movementHandlers.onToBottomDeceleration(
+                            -1 * ctx.lastApproximateVelocity,
+                            runOnUIPlatformSelect({
+                                android: 70,
+                                default: 50,
```

Comment:
Probably you need to put all the values in constants and use `// @inline`. This also applies to other modules.

Reply:
I actually don't think that it have to be moved to constants as it's not gonna be re-used, actually it might be bad to re-use them, as it should be applied only here, and If someone was changed it, it would break the carefully chosen behaviour :) It's like you don't want to move second parameter of `setTimeout` to constants, until it's a common case :)